### ISSUE TITLE
Add acceptance tests for more variety of user Ids

### ISSUE THREAD

**Issue by phil-davis** [owncloud/core#33596]
Current acceptance tests user user IDs like ``user0``, ``user1``,... There are some tests that verify that user IDs with non-alpha-numeric characters can be created (or not created). But we do not have much verification that user IDs containing those non-alpha-numeric characters actually work correctly in a range of situations.

- [ ] run some or all acceptance tests with "interesting" user IDs
- [ ] run some or all acceptance tests with user IDs that look like an email address, and with the user's email address set to the same string, and to a different string

We have already "abstracted out" the "standard passwords" used in the acceptance tests - so that the passwords used by the tests can be passed in from environment variables. That allows the acceptance test suite to be run with a different set of interesting/varied passwords if desired.

We could do the same with the user IDs. Then we can easily run the acceptance test suite with user IDs like ``123.Me.test_user@test-organization.org`` (which uses lots of the valid special characters for a user ID).

Note: the webUI message when specifying an invalid user ID says:
```
Error creating user: Only the following characters are allowed in a username: "a-z", "A-Z", "0-9", and "_.@-'"
```
That is a clue to what should be valid and tested.

Note: since core 10.4 `+` is also allowed in a username.

**Comment by phil-davis** [comment]
Related issue that triggered me to write this acceptance test issue: #33594

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks possibly related issues are https://github.com/owncloud/core/issues/22485 (test), https://github.com/owncloud/core/issues/22492 (test), https://github.com/owncloud/core/issues/31579 (test), https://github.com/owncloud/core/pull/32347 (Add getters for private vars of acceptance test traits), and https://github.com/owncloud/core/pull/33506 (Adding more federation acceptance tests).

**Comment by phil-davis** [comment]
- [ ] sort out if a space character should be allowed in a user ID
- [ ] add acceptance tests to verify that a space character is allowed and works (or not)

https://github.com/owncloud/core/issues/33594#issuecomment-440308620

@PVince81 @DeepDiver1975 @patrickjahns any idea if a user ID like ``John Smith`` should be valid?

**Comment by phil-davis** [comment]
Added to upcoming sprint just for the purpose of adding basic validation tests for whatever should be the allowed and disallowed user ID strings, when attempts are made to create such user IDs with each of
- [ ] occ command
- [ ] Provisioning API
- [ ] webUI user management page

Sorting out how to run all acceptance tests with varying/unusual user names will need more thought/discussion and time to implement.

**Comment by phil-davis** [comment]
Another issue/regression that happened with numeric UIDs and in a particular UI+API call: #37324 

We need to think about how we can do this - to "optionally" run whole sets of UI and/or API tests with a variety of UIDs/Dispaly Names and other variables.

**Comment by phil-davis** [comment]
Code is merged in the PRs linked above.
PR https://github.com/owncloud/core/pull/37506 is an example of a test run with different user names and other user attributes.

ToDo:
- [ ] add to the acceptance test documentation about how to make use of this.

**Comment by phil-davis** [comment]
This can be closed when #38364 is merged.
